Modix.Services/Utilities/EnumerableExtensions.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System.Collections.Generic;
+
+namespace Modix.Services.Utilities
+{
+    public static class EnumerableExtensions
+    {
+        public static IEnumerable<(T Value, int Index)> AsIndexable<T>(this IEnumerable<T> source)
+        {
+            var index = 0;
+
+            foreach (var item in source)
+            {
+                yield return (item, index);
+                index++;
+            }
+        }
+
+        public static IEnumerable<(TFirst First, TSecond Second)> ZipOrDefault<TFirst, TSecond>(this IEnumerable<TFirst> first, IEnumerable<TSecond> second)
+        {
+            using (var e1 = first.GetEnumerator())
+            using (var e2 = second.GetEnumerator())
+            {
+                while (true)
+                {
+                    var e1Moved = e1.MoveNext();
+                    var e2Moved = e2.MoveNext();
+
+                    if (!e1Moved && !e2Moved)
+                        break;
+
+                    yield return
```

Comment:
Following up on my other comment, this would then be 

```suggestion
                    yield return
                    (
                        e1.Current ?? default, 
                        e2.Current ?? default
                    );
```

Reply:
Wouldn't that just be a no-op?

Reply:
...No?

Reply:
What makes you think that?

Reply:
If T is a class and e.Current is null, it'll just resolve to `default`... which is `null`.
If T is a struct, then e.Current will never be `null`